sdk/samples/iot/pnp/pnp_protocol.h

```diff
@@ -13,14 +13,22 @@
 #include <azure/core/az_span.h>
 #include <azure/iot/az_iot_hub_client.h>
 
+typedef enum
+{
+  PNP_METHODS_PUBLISH_TOPIC,
+  PNP_TELEMETRY_PUBLISH_TOPIC,
+  PNP_TWIN_DOCUMENT_PUBLISH_TOPIC,
+  PNP_TWIN_PATCH_PUBLISH_TOPIC
+} pnp_publish_topic;
```

Comment:
What is this used for?

Reply:
I consolidated all the publish topics into a single set_publish_topic function to eliminate some of the duplicated code everywhere for getting the publish topic.  Also an attempt to follow along with the publish_mqtt_message and not have az_result returns to check in so many places.  

With talk today on removing az_result from returning in general, this may no longer be as helpful.

Reply:
I'm not so sure about this one. I can't picture what this would be used for and it might be trying to abstract too much from the user (sort of creating another API layer).